assets/src/edit-story/app/media/mediaProvider.js

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import useLoadMedia from './actions/useLoadMedia';
+import useCompleteMedia from './actions/useCompleteMedia';
+import useReloadMedia from './actions/useReloadMedia';
+import useResetMedia from './actions/useResetMedia';
+import Context from './context';
+
+function MediaProvider( { children } ) {
+	const [ media, setMedia ] = useState( [] );
+	const [ mediaType, setMediaType ] = useState( '' );
+	const [ searchTerm, setSearchTerm ] = useState( '' );
+	const [ isMediaLoaded, setIsMediaLoaded ] = useState( false );
+	const [ isMediaLoading, setIsMediaLoading ] = useState( false );
```

Comment:
I think it would be a lot cleaner with a `useReducer` rather than these 5 separate but still interlinked states. It is however a rather big rewrite, I can approve without it if we really need it in now. Maybe just make a tech debt ticket about it in case we want to proceed as is.

Reply:
I agree, we need to use reducer here, there is going be more stateless entries added here when infinite scroll #23 ( page number ) and the upload progress #231 are added. These refactor at that point.